torch/onnx/verification.py

```diff
@@ -243,6 +246,219 @@ def compare_ort_pytorch_model_with_input(input_args, input_kwargs):
+        raise ValueError("Need at least two set of test inputs to compare.")
+
+    # TODO: refactor utils.py to remove duplicated code of context setup.
+    opset_version = export_options.opset_version
+    operator_export_type = export_options.operator_export_type
+    export_modules_as_functions = export_options.export_modules_as_functions
+    training = export_options.training
+    verbose = export_options.verbose
+    dynamic_axes = export_options.dynamic_axes
+    input_names = export_options.input_names
+    output_names = export_options.output_names
+
+    if opset_version is None:
+        opset_version = _constants.onnx_default_opset
+
+    export_modules_as_functions = utils._setup_trace_module_map(
+        model, export_modules_as_functions
+    )
+
+    if not operator_export_type:
+        if _C_onnx._CAFFE2_ATEN_FALLBACK:
+            operator_export_type = _C_onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK
+        else:
+            operator_export_type = _C_onnx.OperatorExportTypes.ONNX
+
+    _globals.GLOBALS.export_onnx_opset_version = opset_version
+    _globals.GLOBALS.operator_export_type = operator_export_type
+
+    with utils.exporter_context(model, training, verbose):
+        val_do_constant_folding = utils._decide_constant_folding(
+            export_options.do_constant_folding, operator_export_type, training
+        )
+
+        if dynamic_axes is None:
+            dynamic_axes = {}
+        utils._validate_dynamic_axes(dynamic_axes, model, input_names, output_names)
+
+        ref_onnx_graph = None
+        for arg, kwargs in test_input_sets:
+            export_inputs = _prepare_input_for_export(arg, kwargs)
+            export_inputs = utils._decide_input_format(model, export_inputs)
+            onnx_graph, _, _ = utils._model_to_graph(
+                model,
+                export_inputs,
+                verbose,
+                input_names,
+                output_names,
+                operator_export_type,
+                val_do_constant_folding,
+                training=training,
+                dynamic_axes=dynamic_axes,
+            )
+
+            if ref_onnx_graph is None:
+                ref_onnx_graph = onnx_graph
+                continue
+
+            graph_diff_report = _GraphDiff(ref_onnx_graph, onnx_graph).diff_report()
+            if graph_diff_report:
+                return graph_diff_report
+
+    return ""
+
+
+def check_export_model_diff(
+    model: Union[torch.nn.Module, torch.jit.ScriptModule],
+    test_input_sets: Sequence[Tuple[Tuple[Any, ...], Mapping[str, Any]]],
+    export_options: Optional[_experimental.ExportOptions] = None,
+) -> str:
+    """Verify exported model discrepancy between different sets of inputs.
+
+    A graph is exported for each set of inputs. The exported graphs are then compared
+    to each other, and discrepancies of first pair of nodes are reported. This function
+    first checks the jit graph, and then the onnx graph.
```

Comment:
I would mention that the function returns the jit graph if it finds any discrepancy, without bothering to look at the onnx graph (but phrased better)